src/details/ArborX_DetailsDistributedTreeImpl.hpp

```diff
@@ -42,6 +42,70 @@ struct DefaultCallbackWithRank
   }
 };
 
+template <class Predicates, class Distances>
+struct WithinDistanceFromPredicates
+{
+  Predicates predicates;
+  Distances distances;
+};
+
+} // namespace Details
+
+template <class Predicates, class Distances>
+struct AccessTraits<
+    Details::WithinDistanceFromPredicates<Predicates, Distances>, PredicatesTag>
+{
+  using Access = AccessTraits<Predicates, PredicatesTag>;
+  using Predicate = typename Details::AccessTraitsHelper<Access>::type;
+  using Geometry =
+      std::decay_t<decltype(getGeometry(std::declval<Predicate const &>()))>;
+  using Self = Details::WithinDistanceFromPredicates<Predicates, Distances>;
+
+  using memory_space = typename Access::memory_space;
+  using size_type = size_t;
```

Comment:
Shouldn't this be the return type of `Access::size(predicates)`?

Reply:
I don't think it matters much but I can change if you like